src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsFrameworkCompatibility.cs

```diff
@@ -91,7 +91,17 @@ public FrameworkName GetNearest(FrameworkName targetFramework, IEnumerable<Frame
                 return null;
             }
 
-            return new FrameworkName(nearest.DotNetFrameworkName);
+            try
+            {
+                return new FrameworkName(nearest.DotNetFrameworkName);
+            }
+            catch (System.ArgumentException)
+            {
+                // GetNearest isn't supported for net5-era frameworks, since
+                // their identifiers don't losslessly translate to
+                // `FrameworkName`s.
+                return null;
```

Comment:
Aren't all net5.0-* TFMs the .NETCoreApp framework name?

Even if it doesn't allow us to distinguish between "net5.0" and "net5.0-android", I'm scared that random VS components will break with .NET 5 projects if we return null here.

Reply:
Shall we just let this crash, then?